_posts/2022-09-08-storage-write-removal-before-conditional-termination.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: post
+published: true
+title: 'Optimizer Bug Causing the Removal of Storage Writes Before Calls to Functions Conditionally Terminating the Transaction'
+date: '2022-09-08'
+author: Solidity Team
+category: Security Alerts
+---
+On September 5, 2022, a bug in Solidity's Yul optimizer was found by differential fuzzing.
+
+The bug was introduced in version 0.8.13 and [Solidity version 0.8.17](https://github.com/ethereum/solidity/releases/tag/v0.8.17), released on September 07, 2022, provides a fix. The bug is significantly easier to trigger with optimized via-IR code generation, but can theoretically also occur in optimized legacy code generation.
+
+We assigned the bug a severity of "medium/high".
+
+
+## Who Should Be Concerned
+
+If you're using optimized legacy code generation, you only need to be concerned, if you use large inline assembly blocks that involve user-defined assembly functions that involve the ``return(...)`` or ``stop()`` instructions. Furthermore only inline assembly blocks that do not interact with any surrounding Solidity variables can be affected.
+
+If you're using optimized via-IR code generation, you are more likely to be affected:
+
+You should carefully check your contract whenever it involves any code path that can write to storage and that can then either continue to an early termination using assembly ``return(...)`` or ``stop()``, or can bypass this early termination. This includes cases in which the storage writes or the early termination occur in complex control flow or in arbitrarily nested function calls.
```

Comment:
This sounded a bit ambiguous to me regarding whether you need both at the same time or just one of them. How about this?
```suggestion
You should carefully check your contract whenever it involves any code path that first writes to storage and then continues to an early termination using assembly ``return(...)`` or ``stop()``, but can also conditionally bypass this early termination. This includes cases in which the storage writes or the early termination occur in complex control flow or in arbitrarily nested function calls.
```